### ISSUE TITLE
std::bad_alloc when inserting from pyarrow with fixed size array

### ISSUE THREAD

**Issue by eaubin** [duckdb/duckdb#3813]
#### What happens?

Inserting from a big enough pyarrow table with a fixed size array will cause the process with exit with 'std::bad_alloc'

#### To Reproduce

```python
import pandas as pd
import pyarrow as pa
import duckdb
import numpy as np

n = 8888 # Smaller sizes work
df = pd.DataFrame.from_dict({'xs':pd.Series([np.random.random((100)) for i in range(n)])})
tbl = pa.Table.from_pandas(df,
         pa.schema([('xs',pa.list_(pa.float32()
                                   , list_size=100 # When list_size is omitted, the process does not exit
                                  ))]))
con=duckdb.connect(':memory:')
con.execute("""CREATE TABLE tofill (xs FLOAT[])""")
con.execute("INSERT INTO tofill (xs) SELECT xs FROM tbl");
```

With the latest version, it prints `free(): invalid next size (fast)` before exiting.

#### Environment (please complete the following information):
 - OS: Windows
 - DuckDB Version: 0.3.5.dev1091
 - DuckDB Client: python

#### Before Submitting

- [X] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [X] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

**Comment by Mytherin** [comment]
Also seems to happen without the insert, so this is likely related to the Arrow scan:

```py
import pandas as pd
import pyarrow as pa
import duckdb
import numpy as np

n = 8888 # Smaller sizes work
df = pd.DataFrame.from_dict({'xs':pd.Series([np.random.random((100)) for i in range(n)])})
tbl = pa.Table.from_pandas(df,
         pa.schema([('xs',pa.list_(pa.float32()
                                   , list_size=100 # When list_size is omitted, the process does not exit
                                  ))]))
con=duckdb.connect(':memory:')
con.execute("SELECT xs FROM tbl").df();
```

@pdet can you have a look?

**Comment by Tishj** [comment]
I haven't figured out the issue precisely yet, but it's caused by a bug in arrow_conversion.cpp:ArrowToDuckDBList
```c++
	if (original_type.first == ArrowVariableSizeType::FIXED_SIZE) {
		//! Have to check validity mask before setting this up
		idx_t offset = (scan_state.chunk_offset + array.offset) * original_type.second;
		if (nested_offset != -1) {
			offset = original_type.second * nested_offset;
		}
		start_offset = offset;
		auto list_data = FlatVector::GetData<list_entry_t>(vector);
		for (idx_t i = 0; i < size; i++) {
			auto &le = list_data[i];
			le.offset = cur_offset;
			le.length = original_type.second;
			cur_offset += original_type.second;
		}
		list_size = cur_offset;
```
When specifying the list_size, the second time around the `list_size` variable is 102400 and the `start_offset` is also 102400.
which causes `list_size -= start_offset;` to be 0, and we pass this value to `ListVector::SetListSize()` so the Verify assertion fails

Without specifying the list_size parameter in `pa.list_()` the `list_size` variable is 204800 and the `start_offset` is 102400

**Comment by Tishj** [comment]
I took a very naive approach and changed
`list_size = cur_offset;` -> `list_size = start_offset + cur_offset;`

This seems to have fixed the issue in this case, @pdet are there any intricacies I should be aware of that might make this fix not work in some cases?

**Comment by pdet** [comment]
Yeah, that looks about right. Did you push this fix already @Tishj ?

**Comment by Tishj** [comment]
This issue should be fixed by #4088, closing this
